### PR TITLE
issues/863 - fix keyring slots

### PR DESCRIPTION
**Pull request by RichardGe**
fix for   https://github.com/cmangos/issues/issues/863

### CODE REVIEW

File: src/game/Entities/Player.cpp [hunk 1]
```diff
@@ -10514,6 +10514,24 @@ void Player::RemoveItemFromBuyBackSlot(uint32 slot, bool del)
     }
 }
 
+uint32 Player::GetMaxKeyringSize() const 
+{ 
+	int playerLevel = getLevel();
+	if ( playerLevel <= PLAYER_KEYRING_SIZE_STEP1_LVLMAX )
```
Comment (killerwife) [inline src/game/Entities/Player.cpp]: Why not static array? divide by 10, 6 numbers, single access @cyberium what do you think

Comment (RichardGe) [reply src/game/Entities/Player.cpp]: why not: something like :  
[0] = 4   // level : 0 -> 9
[1] = 4   // 10 -> 19
[2] = 4   // 20 -> 29
[3] = 4   // 30 -> 39
[4] = 8   // 40 -> 49
[5] = 12  // 50 -> 59
[6] = 12  //  level >= 60

btw, even if this code is only for Classic, I don't know how the slots evolve on TBC, ...etc...

Comment (killerwife) [reply src/game/Entities/Player.cpp]: Afaik this isnt sent to client, so you can check tbc or wotlk by leveling up char and watching client UI.

Comment (RichardGe) [reply src/game/Entities/Player.cpp]: I only have Classic on my side, I'll improve this commit with static array as I have described. later this week.

**Comment by RichardGe** [comment]
modified as @killerwife suggested, with a static array.
I have also renamed GetMaxKeyringSize -->  GetMaxKeyringClientSize, in order to avoid confusion between the size seen by server (always 16 slots), and the size seen by Client (4, 8 or 12 slots)

File: src/game/Entities/Player.cpp (hunk 1 above)
Comment (cala) [reply src/game/Entities/Player.cpp]: For TBC and Wotlk,, it was changed in patch 2.0.1 to become a dynamic size keyring (so size no longer based on level). At time of patches 2.4.3 and 3.3.5, the max size of the keyring was 32 slots.

**Comment by cala** [comment]
Merged, thank you very much!

**Pull request closed without merging**